File: Ruby/civitas/pkg/civitas-0.0.1/lib/sorpresa_pagar_cobrar.rb
#encoding: utf­8
require_relative "sorpresa.rb"
module Civitas
class SorpresaPagarCobrar < Sorpresa
  attr_accessor :valor
  def initialize(_valor, _texto)
    super(_texto)
    @valor = _valor
  end
  
  def aplicar_a_jugador(_actual, _todos)
    if(jugador_correcto(_actual, _todos))
      informe(_actual, _todos)
      _todos.at(_actual).modificar_saldo(@valor)
    end
  end
  
  def to_ss()
    return @texto
  end
end
end

File: Java/civitas/src/civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;

public class SorpresaPagarCobrar extends Sorpresa {
    private int valor;
    
    SorpresaPagarCobrar(int _valor, String _texto){
        super(_texto);
        valor = _valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            this.informe(actual, todos);
            todos.get(actual).modificarSaldo(this.valor);
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa{ " + super.getTexto() + ", valor " + Integer.toString(valor) + " }";          
    }
}
